sycl/include/sycl/ext/oneapi/experimental/invoke_simd.hpp

```diff
@@ -331,6 +345,42 @@ template <typename T>
 using remove_ref_from_func_ptr_ref_type_t =
     typename remove_ref_from_func_ptr_ref_type<T>::type;
 
+template <typename T> struct strip_regcall_from_function_ptr;
+
+template <typename Ret, typename... Args>
+struct strip_regcall_from_function_ptr<Ret (*)(Args...)> {
+  using type = Ret (*)(Args...);
+};
+
+template <typename Ret, typename... Args>
+struct strip_regcall_from_function_ptr<Ret(__regcall *)(Args...)> {
+  using type = Ret (*)(Args...);
+};
+
+template <typename T>
+using strip_regcall_from_function_ptr_t =
+    typename strip_regcall_from_function_ptr<T>::type;
+
+template <typename Ret, typename... Args>
+constexpr bool has_ref_arg(Ret (*)(Args...)) {
+  return std::is_reference_v<Ret> || (... || std::is_reference_v<Args>);
+}
```

Comment:
This function is named 'has_ref_arg', while it check both return and args. It could be better to name it 'has_ref_ret_or_arg'.
But even better (IMO) would be to have 2 functions:
```suggestion
template <typename Ret, typename... Args>
constexpr bool has_ref_ret(Ret (*)(Args...)) {
  return std::is_reference_v<Ret>;
}

template <typename Ret, typename... Args>
constexpr bool has_ref_arg(Ret (*)(Args...)) {
  return false || (... || std::is_reference_v<Args>);
}
```

Reply:
this is a good idea, thanks. let me implement this.